### PR TITLE
document node17 compatibility

### PR DESCRIPTION
**Pull request by mo-fc**
[migration-guides](https://deploy-preview-395--heuristic-almeida-1a1f35.netlify.app/docs/appendices/migration-guides.html)

[installation](https://deploy-preview-395--heuristic-almeida-1a1f35.netlify.app/docs/essentials/installation.html)

### CODE REVIEW

File: source/docs/appendices/migration-guides.md
Comment (abailleulfc) [inline source/docs/appendices/migration-guides.md:14]:
```suggestion
We now support node versions 14, 16, 17. If you are using node 12 you will need to update your node version (node 16 is recommended).
```

File: source/docs/appendices/migration-guides.md
Comment (abailleulfc) [inline source/docs/appendices/migration-guides.md:18]: I am unsure it is necessary to detail the reasons here :thinking: 

```suggestion
- ``@apollo/*` are only installed as sub-dependencies and therefore **should not be used**. In `2.15.0` you need to replace all uses of `@apollo/*` in your codebase with `react-apollo`, or `graphql-tag` imports where applicable. Check [here](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1215/diffs?commit_id=8fae8438d46200cbd310b16c30b3078da28b7d42) and [here](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1215/diffs?commit_id=418a291f917660b18672b6be2eb2778cb51dfef9)/ e.g:
```

File: source/docs/appendices/migration-guides.md
Comment (abailleulfc) [inline source/docs/appendices/migration-guides.md:25]:
```suggestion
- Resolvers for fields not in the `schema` now throw errors. To avoid this remove all unneeded `field`/`Type` resolvers that do not have a representation in the GraphQL `schema`. Check our [cleaning commit](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1216/diffs?commit_id=1e2e04e7f28dba2ebd36da50701478a13a58a238) regarding this issue.
```

File: source/docs/appendices/migration-guides.md
Comment (abailleulfc) [inline source/docs/appendices/migration-guides.md:29]: Are this line and the previous one  also documented in the readme and on the [installation documentation page](https://developers.front-commerce.com/docs/essentials/installation.html#Installation-using-the-skeleton) ?

**Review by abailleulfc — CHANGES_REQUESTED** [review]

File: source/docs/appendices/migration-guides.md
Comment (mo-fc) [reply source/docs/appendices/migration-guides.md:29]: good point

Comment (mo-fc) [reply source/docs/appendices/migration-guides.md:29]: done please check

**Review by abailleulfc — APPROVED** [review]
Seems go to me, maybe wait for a second review before merging

File: source/docs/essentials/installation.md
Comment (real34) [inline source/docs/essentials/installation.md:137]: **suggestion:** can't we change the `start` script to force this value?\
I.e: the `bin/front-commerce` script is used to start the app. Can't it do it itself depending on the node version?

Typo
```suggestion
  NODE_OPTIONS=--openssl-legacy-provider npm run start
```

File: source/docs/essentials/installation.md
Comment (real34) [inline source/docs/essentials/installation.md:169]: See comment above about this

File: source/docs/appendices/migration-guides.md
Comment (real34) [inline source/docs/appendices/migration-guides.md:14]:
```suggestion
We now support node versions 14, 16, 17. If you are using node 12 ([which reached it's end of life](https://nodejs.org/en/about/releases/)), you must update your node version (we recommend node 16).
```

File: source/docs/appendices/migration-guides.md
Comment (real34) [inline source/docs/appendices/migration-guides.md:16]:
```suggestion
To support these latest node versions, we had to ugrade some dependencies. These upgrades imply changes that could impact your codebase:
```

File: source/docs/appendices/migration-guides.md
Comment (real34) [inline source/docs/appendices/migration-guides.md:18]:
```suggestion
- ``@apollo/*` are only installed as sub-dependencies and therefore **should not be used directly**. In `2.15.0` you need to replace all uses of `@apollo/*` in your codebase with `react-apollo`, or `graphql-tag` imports where applicable. Check [here](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1215/diffs?commit_id=8fae8438d46200cbd310b16c30b3078da28b7d42) and [here](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1215/diffs?commit_id=418a291f917660b18672b6be2eb2778cb51dfef9)/ e.g:
```

File: source/docs/appendices/migration-guides.md
Comment (real34) [inline source/docs/appendices/migration-guides.md:28]:
```suggestion
- When using node 17 you need to prefix your npm commands by `NODE_OPTIONS=--openssl-legacy-provider` e.g. `NODE_OPTIONS=--openssl-legacy-provider npm start`, `NODE_OPTIONS=--openssl-legacy-provider npm run build`.
```

File: source/docs/essentials/installation.md
Comment (real34) [reply source/docs/essentials/installation.md:137]: If we don't find a solution, we could add a blockquote explaining that it currently is needed because FC is still using webpack 4, and redirecting to https://github.com/webpack/webpack/issues/13572#issuecomment-1011497486 for more details.

Comment (paul-vd) [reply source/docs/essentials/installation.md:137]: @mo-fc The `processs` hould include the node version so we could just do this in the bin right?

This is from the bin/front-commerce when running in debugger you can see all the available things on `process`

```bash
// process.versions
ares:'1.18.1'
brotli:'1.0.9'
cldr:'40.0'
icu:'70.1'
llhttp:'6.0.4'
modules:'93'
napi:'8'
nghttp2:'1.45.1'
nghttp3:'0.1.0-DEV'
ngtcp2:'0.1.0-DEV'
node:'16.14.2'
openssl:'1.1.1n+quic'
tz:'2021a3'
unicode:'14.0'
uv:'1.43.0'
v8:'9.4.146.24-node.20'
zlib:'1.2.11'
```

Code - front-commerce — front-commerce 2022-05-06 at 10 37 25

**Review by paul-vd — CHANGES_REQUESTED** [review]

File: source/docs/appendices/migration-guides.md
Comment (dpobel) [reply source/docs/appendices/migration-guides.md:14]: our CI allows failures for node 17, so I don't think we can write that we support node 17

Comment (mo-fc) [reply source/docs/appendices/migration-guides.md:14]: it allows but it is not failing

**Comment by mo-fc** [comment]
@paul-vd [regarding](https://github.com/front-commerce/developers.front-commerce.com/pull/395#discussion_r866612622) I was working more on the feature/SPOC more before now I got the time for review :/ its too late now we can include it in next release

File: source/docs/essentials/installation.md
Comment (mo-fc) [reply source/docs/essentials/installation.md]: I think we shouldnt mention we are still using webpack 4 on our installation guide. its like a bad omen

File: source/docs/appendices/migration-guides.md
Comment (dpobel) [inline source/docs/appendices/migration-guides.md:62]: this one can not be overriden

File: source/docs/appendices/migration-guides.md
Comment (dpobel) [inline source/docs/appendices/migration-guides.md:74]: same

File: source/docs/appendices/migration-guides.md
Comment (dpobel) [inline source/docs/appendices/migration-guides.md:514]: empty line not needed

File: source/docs/appendices/migration-guides.md
Comment (dpobel) [inline source/docs/appendices/migration-guides.md:58]: a table for non tabular data? come on we are in 2022

File: source/docs/essentials/installation.md
Comment (dpobel) [inline source/docs/essentials/installation.md:135]: I still don't get why we put so much content in our documentation for a version that will be unsupported in the end of the month and probably not used by anybody.

File: source/docs/appendices/migration-guides.md
Comment (Kagaminara) [inline source/docs/appendices/migration-guides.md:515]: This first sentense is missing a verb I guess? It might be only me though, but I would add this :
```suggestion
- Resolvers for fields that are not in the `schema` now throw errors. To avoid this remove all unneeded `field`/`Type` resolvers that do not have a representation in the GraphQL `schema`. Check our [cleaning commit](https://gitlab.com/front-commerce/front-commerce/-/merge_requests/1216/diffs?commit_id=1e2e04e7f28dba2ebd36da50701478a13a58a238) regarding this issue.
```

File: source/docs/appendices/migration-guides.md
Comment (Kagaminara) [inline source/docs/appendices/migration-guides.md:516]:
```suggestion
- `makeApolloClientStub` now expects resolvers as an argument instead of mocks. This is in tandem with the [`graphql-tools` upgrade notes](https://www.graphql-tools.com/docs/mocking#migration-from-v7-and-below). Note that we provided an adapter that converts `mocks` to `resolvers` so that old tests are not affected. However it is recommended to make the change. To disable the adapter's warning set `FRONT_COMMERCE_WEB_PRINT_MAKEAPOLLOCLIENTSTUB_MOCKS_DEPRECATION_WARNING` environment variable to `"false"`.
```

File: source/docs/appendices/migration-guides.md
Comment (Kagaminara) [inline source/docs/appendices/migration-guides.md:541]: Now we add the problem with dependencies not updating because of the `package-lock.json`, I wonder if we should add `rm package-lock.json` in those instructions (for npm > 8.5~ish)? It doesn't look like a good idea, but just `npm install` doesn't seem to do much now (with or without removing `node_modules`).

File: source/docs/appendices/migration-guides.md
Comment (mo-fc) [reply source/docs/appendices/migration-guides.md:53]:
```suggestion
- `@apollo/*` are only installed as sub-dependencies and therefore **should not be used directly**. In `2.15.0` you need to replace all uses of `@apollo/*` in your codebase with `react-apollo`, or `graphql-tag` imports where applicable.
```

File: source/docs/appendices/migration-guides.md
Comment (mo-fc) [reply source/docs/appendices/migration-guides.md:541]: yes I put the `rm -rf ./node_modules/` which deletes the node modules followed by `npm install`

File: source/docs/essentials/installation.md
Comment (mo-fc) [reply source/docs/essentials/installation.md:135]: this will be the same with node 18 which is covered by the >=17 remark

File: source/docs/appendices/migration-guides.md
Comment (Kagaminara) [reply source/docs/appendices/migration-guides.md]: Not the `node_modules`, I was talking about removing the `package-lock.json` directly

**Comment by mo-fc** [comment]
Ill merge now any further updates can be patched later

**Review by dpobel — APPROVED** [review]

**Pull request merged**
